### PR TITLE
Improved output if the sdk check fails

### PR DESCRIPTION
**Pull request by JCash**
We improved the output in case of missing sdk paths.

You can do this check manually using:
`./scripts/build.py check_sdk --platform=arm64-android`

If it succeeds it won't print anything. You can however make it verbose:
`./scripts/build.py check_sdk --platform=arm64-android --verbose`
and it will output something like:
```
SDK info:
{'api': '21',
 'build_tools': '/Users/mawe/Library/android/sdk/build-tools/35.0.0',
 'ndk': '/Users/mawe/Library/android/sdk/ndk/25.2.9519653',
 'sdk': '/Users/mawe/Library/android/sdk',
 'version': '35.0.0'}
'check_sdk' completed in 0.00 s
```

## PR checklist

* [ ] Code
	* [ ] Add engine and/or editor unit tests.
	* [ ] New and changed code follows the overall code style of existing code
	* [ ] Add comments where needed
* [ ] Documentation
	* [ ] Make sure that API documentation is updated in code comments
	* [ ] Make sure that manuals are updated (in github.com/defold/doc)
* [ ] Prepare pull request and affected issue for automatic release notes generator
	* [ ] Pull request - Write a message that explains what this pull request does. What was the problem? How was it solved? What are the changes to APIs or the new APIs introduced? This message will be used in the generated release notes. Make sure it is well written and understandable for a user of Defold.
	* [ ] Pull request - Write a pull request title that in a sentence summarises what the pull request does. Do not include "Issue-1234 ..." in the title. This text will be used in the generated release notes.
	* [ ] Pull request - Link the pull request to the issue(s) it is closing. Use on of the [approved closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).
	* [ ] Affected issue - Assign the issue to a project. Do not assign the pull request to a project if there is an issue which the pull request closes.
	* [ ] Affected issue - Assign the "breaking change" label to the issue if introducing a breaking change.
	* [ ] Affected issue - Assign the "skip release notes" is the issue should not be included in the generated release notes.

----------

Example of a well written PR description:

1. Start with the user facing changes. This will end up in the release notes.
1. Add one of the GitHub approved closing keywords
1. Optionally also add the technical changes made. This is information that might help the reviewer. It will not show up in the release notes. Technical changes are identified by a line starting with one of these:
   1. `### Technical changes` 
   1. `Technical changes:`
   2. `Technical notes:`

```
There was a anomaly in the carbon chroniton propeller, introduced in version 8.10.2. This fix will make sure to reset the phaser collector on application startup.

Fixes #1234

### Technical changes
* Pay special attention to line 23 of phaser_collector.clj as it contains some interesting optimizations
* The propeller code was not taking into account a negative phase.
```

### CODE REVIEW

File: README_BUILD.md [hunk 1]
```diff
@@ -8,17 +8,16 @@ Make sure you have followed the [setup guide](/README_SETUP.md) before attemptin
 
 ## IMPORTANT PREREQUISITE - PLATFORM SDKs!
 
-### Using local installation
+Our build setup can find the installation of the platform SDK's for the common platforms.
+
+See the setup for the [Platform SDK](./README_SETUP.md#platform-sdk)
 
-Our build setup can find the installation the platform SDK's for these platforms
+<details><summary>Locally installed platform sdks</summary><p>
```
Comment (JCash) [inline README_BUILD.md:15]: I opted to minimize the wall-of-text to make it slightly more welcoming (I hope)

File: README_BUILD.md [hunk 2]
```diff
@@ -8,17 +8,16 @@ Make sure you have followed the [setup guide](/README_SETUP.md) before attemptin
...
-* Windows - Visual Studio + clang++
-* macOS/iOS - XCode
-* Android - Android Studio
-* Linux - clang++
-* Consoles - The vendor specific sdks
+### Using local installation
 
-In the future, we want to support HTML5 as well, for easier setup for all platforms.
+Our build setup can find the installation of the platform SDK's for the common platforms.
+See [Platform SDK](./README_SETUP.md#platform-sdk)
```
Comment (JCash) [inline README_BUILD.md:20]: I think it's part of the setup, so I refer to that document instead.

File: README_BUILD.md [hunk 3]
```diff
@@ -44,6 +45,43 @@ The standard workflow when building the engine is the following:
 
 When working on a new feature or fixing a bug you start by first building the engine once as described above. You then proceed to develop your feature or fix the bug and rebuild and test changes until satisfied. When you do a rebuild you can speed things up by only building the parts that have changed.
 
+## Build examples
```
Comment (JCash) [inline README_BUILD.md:48]: To get to the "good stuff" a little faster, I present a few examples upfront.
Currently visible on the first page when visiting this readme.

File: README_SETUP.md [hunk 4]
```diff
@@ -4,6 +4,18 @@
...
+
+To make contributions easier, we detect local installations of the common platform SDK's:
+
+* macOS + iOS: [XCode](https://apps.apple.com/us/app/xcode/id497799835)
+* Linux: [Clang++](https://apt.llvm.org/)
+* Windows: [Visual Studio COmmunity 2022](https://visualstudio.microsoft.com/vs/older-downloads/)
+  * (Note that we also require Clang!)
+* Android: [Android Studio](https://developer.android.com/studio)
+
+For consoles, we refer to each vendor's installation notes.
```
Comment (JCash) [inline README_SETUP.md:17]: Since these are kind of the base sdk, I think it's good to start with.

File: README_SETUP.md [hunk 5]
```diff
@@ -247,17 +257,18 @@ It is recommended but not required that you install the following software:
...
 
-* **wget** + **curl** - for downloading packages
+* **wget** + **curl** - for downloading packages (used for downloading packages in different scripts)
 * **7z** - for extracting packages (archives and binaries)
-* **ccache** - for faster compilations of source code
+* **ccache** - for faster compilations of source code (optional)
 * **cmake** for easier building of external projects
 * **patch** for easier patching on windows (when building external projects)
 * **ripgrep** for faster search
+* **dos2unix** tool to convert line endings of certain source files (e.g. when building files in `share/ext`)
```
Comment (JCash) [inline README_SETUP.md:266]: Just as optional as the others. Used rarely.

File: build_tools/sdk.py [hunk 6]
```diff
@@ -111,6 +111,15 @@
...
 defold_info['x86_64-linux']['pattern'] = 'linux/clang-%s' % VERSION_LINUX_CLANG
 
+## **********************************************************************************************
+
+def log_verbose(verbose, msg):
+    if verbose:
+        log.log(msg)
+
+class SDKException(Exception):
+    pass
```
Comment (JCash) [inline build_tools/sdk.py:121]: Introducing an exception, to make it easier to exit from these functions.

File: scripts/build.py [hunk 7]
```diff
@@ -557,20 +557,26 @@ def get_local_or_remote_file(self, path):
         return path
 
     def check_sdk(self):
+        # TODO: Make sure this check works for all platforms
+        if target_platform in ('js-web', 'wasm-web'): # some platforms are not yet supported using this sdk_info mechanic
+            return
```
Comment (JCash) [inline scripts/build.py:562]: I'm currently working on detecting emscripten as well.

File: scripts/build.py [hunk 8]
```diff
@@ -557,20 +557,26 @@ def get_local_or_remote_file(self, path):
...
-            # TODO: Make sure this check works for all platforms
-            if not self.sdk_info:
-                print("Couldn't find any sdks for platform", target_platform)
-                print("We recommend you follow the setup guide found here: %s" % "https://github.com/defold/defold/blob/dev/README_BUILD.md#important-prerequisite---platform-sdks")
-                sys.exit(1)
+        # TODO: Make sure this check works for all platforms
+        if not self.sdk_info:
+            if not self.verbose:
+                # Do it again, with verbose on, so that we can get more info straight away:
+                sdk.get_sdk_info(sdkfolder, target_platform, True)
```
Comment (JCash) [inline scripts/build.py:572]: The "bug fix":
Make sure the user sees that the sdk wasn't setup correctly.

File: scripts/build.py [hunk 9]
```diff
@@ -2533,6 +2543,18 @@ def _form_env(self):
                       gcloud_keyfile = options.gcloud_keyfile,
                       verbose = options.verbose)
 
+    needs_dynamo_home = True
```
Comment (JCash) [inline scripts/build.py:2546]: another pr

**Review by britzl — APPROVED** [review]

**Pull request merged**